pkg/controller/vpc_nat_gateway.go

```diff
@@ -533,26 +534,21 @@ func (c *Controller) handleUpdateNatGwSubnetRoute(natGwKey string) error {
 		return err
 	}
 	pod := oriPod.DeepCopy()
-	extSubnet, err := c.subnetsLister.Get(util.VpcExternalNet)
-	if err != nil {
-		klog.Errorf("failed to get ovn-vpc-external-network subnet, err: %v", err)
-		return err
-	}
 	var extRules []string
-	if extSubnet.Spec.CIDRBlock != "" && extSubnet.Spec.Gateway != "" {
-		extRules = append(extRules, fmt.Sprintf("%s,%s", extSubnet.Spec.CIDRBlock, extSubnet.Spec.Gateway))
-		if err = c.execNatGwRules(pod, natGwExtSubnetRouteAdd, extRules); err != nil {
-			klog.Errorf("failed to exec nat gateway rule, err: %v", err)
-			return err
-		}
+	var v4ExternalGw, v4InternalGw, v4ExternalCidr string
+	if subnet, ok := c.ipam.Subnets[util.VpcExternalNet]; ok {
+		v4ExternalGw = subnet.V4Gw
+		v4ExternalCidr = subnet.V4CIDR.String()
 	} else {
-		err = fmt.Errorf("failed to get external subnet cidr and gw")
+		return fmt.Errorf("failed to get external subnet %s", util.VpcExternalNet)
+	}
+	extRules = append(extRules, fmt.Sprintf("%s,%s", v4ExternalCidr, v4ExternalGw))
+	if err = c.execNatGwRules(pod, natGwExtSubnetRouteAdd, extRules); err != nil {
+		klog.Errorf("failed to exec nat gateway rule, err: %v", err)
 		return err
 	}
 
-	gwSubnet, err := c.subnetsLister.Get(gw.Spec.Subnet)
-	if err != nil {
-		klog.Errorf("failed to get subnet, err: %v", err)
+	if v4InternalGw, _, err = c.GetGwbySubnet(gw.Spec.Subnet); err != nil {
```

Comment:
should log error here to follow the code style

Reply:
done